scripts/imgtool/image.py

```diff
@@ -222,6 +222,8 @@ def save(self, path, hex_addr=None):
                 padding = bytes([self.erased_val] *
                                 (trailer_size - len(boot_magic))) + boot_magic
                 h.puts(trailer_addr, padding)
+                if self.confirm and not self.overwrite_only:
+                    h[trailer_addr + trailer_size - len(boot_magic) - MAX_ALIGN] = 0x01  # image_ok = 0x01
```

Comment:
What is the problem of adding `image_ok` to an `overwrite_only` image? This code assumes that `IntelHex` allows byte access, wouldn't it be better to just build `padding` correctly based on the `confirm` flag and use `puts()`?

Reply:
I have changed implementation. This `if self.confirm and not self.overwrite_only:` is the same condition as in `def pad_to(self, size):` function, which process bin files.

Reply:
> This if self.confirm and not self.overwrite_only: is the same condition as in def pad_to(self, size): function, which process bin files.

Yeah, there is nothing wrong with it, it's just weird because it is irrelevant if overwite-only mode is being used.